homeassistant/components/sensor/netatmo.py

```diff
@@ -0,0 +1,154 @@
+from homeassistant.const import (CONF_API_KEY, CONF_USERNAME, CONF_PASSWORD,
+                                 TEMP_CELCIUS)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers import validate_config
+from homeassistant.util import Throttle
+
+REQUIREMENTS = [
+    'https://github.com/HydrelioxGitHub/netatmo-api-python/archive/'
+    '59d157d03db0aa167730044667591adea4457ca8.zip'
+    '#lnetatmo==0.3.0.dev1']
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_TYPES = {
+    'temperature': ['Temperature', TEMP_CELCIUS],
+    'co2':         ['CO2', 'ppm'],
+    'pressure':    ['Pressure', 'mb'],
+    'noise':       ['Noise', 'dB'],
+    'humidity':    ['Humidity', '%']
+}
+
+CONF_SECRET_KEY = 'secret_key'
+ATTR_MODULE = 'modules'
+
+# Return cached results if last scan was less then this time ago
+# NetAtmo Data is uploaded to server every 10mn
+# so this time should not be under
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=600)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Get the NetAtmo sensor. """
+
+    if not validate_config({DOMAIN: config},
+                           {DOMAIN: [CONF_API_KEY,
+                                     CONF_USERNAME,
+                                     CONF_PASSWORD,
+                                     CONF_SECRET_KEY]},
+                           _LOGGER):
+        return None
+
+    import lnetatmo
+
+    authorization = lnetatmo.ClientAuth(config.get(CONF_API_KEY, None),
+                                        config.get(CONF_SECRET_KEY, None),
+                                        config.get(CONF_USERNAME, None),
+                                        config.get(CONF_PASSWORD, None))
+
+    if not authorization:
+        _LOGGER.error(
+            "Connection error "
+            "Please check your settings for NatAtmo API.")
+        return False
+
+    data = NetAtmoData(authorization)
+
+    dev = []
+    try:
+        # Iterate each module
+        for module_name, monitored_conditions in config[ATTR_MODULE].items():
+            # Test if module exist """
+            if module_name not in data.get_module_names():
+                _LOGGER.error('Module name: "%s" not found', module_name)
+                continue
+            # Only create sensor for monitored """
+            for variable in monitored_conditions:
+                if variable not in SENSOR_TYPES:
+                    _LOGGER.error('Sensor type: "%s" does not exist', variable)
+                else:
+                    dev.append(
+                        NetAtmoSensor(data, module_name, variable))
+    except KeyError:
+        pass
+
+    add_devices(dev)
+
+
+# pylint: disable=too-few-public-methods
+class NetAtmoSensor(Entity):
+    """ Implements a NetAtmo sensor. """
+
+    def __init__(self, netatmo_data, module_name, sensor_type):
+        self.client_name = 'NetAtmo'
+        self._name = module_name + '_' + SENSOR_TYPES[sensor_type][0]
```

Comment:
- Names do not have to contain underscores but can use spaces instead.
 - Please use string formatting instead of string concatenation:
```python
self.name = "NetAtmo {} {}".format(module_name, SENSOR_TYPES[sensor_type][0])
```